src/lib/core/datetime/native-date-adapter.ts

```diff
@@ -59,18 +60,21 @@ function range<T>(length: number, valueFunction: (index: number) => T): T[] {
 /** Adapts the native JS Date for use with cdk-based components that work with dates. */
 @Injectable()
 export class NativeDateAdapter extends DateAdapter<Date> {
-  constructor(@Optional() @Inject(MAT_DATE_LOCALE) matDateLocale: string) {
-    super();
-    super.setLocale(matDateLocale);
-  }
-
   /**
    * Whether to use `timeZone: 'utc'` with `Intl.DateTimeFormat` when formatting dates.
    * Without this `Intl.DateTimeFormat` sometimes chooses the wrong timeZone, which can throw off
    * the result. (e.g. in the en-US locale `new Date(1800, 7, 14).toLocaleDateString()`
    * will produce `'8/13/1800'`.
    */
-  useUtcForDisplay = true;
+  useUtcForDisplay: boolean;
+
+  constructor(@Optional() @Inject(MAT_DATE_LOCALE) matDateLocale: string, platform: Platform) {
+    super();
+    super.setLocale(matDateLocale);
+
+    // IE does its own time zone correction, so we disable this on IE.
+    this.useUtcForDisplay = !platform.TRIDENT;
```

Comment:
1. Can this be feature-checked (e.g. creating a date and verifying that the time is correct)? 
2. Is it IE-exclusive or does it happen on Edge as well?

Reply:
1. Not really, it depends on the locale the user is in and when / if they have DST
2. IE only

I also intentionally left this public so it can be overridden by users as an easy fix while I investigate any future bugs related to this